app/services/provider_authorisation.rb

```
@@ -126,16 +127,20 @@ def permission_as_training_provider_user?(permission:, course:)
     permission_name = "training_provider_can_#{permission}"
     relationship = relationship_for course: course
 
+    raise RelationshipNotPresent if relationship.blank?
+
     @actor.providers.include?(course.provider) &&
-      (relationship.blank? || relationship.send(permission_name))
+      relationship.send(permission_name)
   end
 
   def permission_as_ratifying_provider_user?(permission:, course:)
     permission_name = "ratifying_provider_can_#{permission}"
     relationship = relationship_for course: course
 
+    return if relationship.blank?
```

can you raise a relationship not present error here as well?